GPU/Common/ShaderId.cpp

```diff
@@ -263,15 +263,15 @@ void ComputeFragmentShaderID(FShaderID *id_out, const Draw::Bugs &bugs) {
 		bool doFlatShading = gstate.getShadeMode() == GE_SHADE_FLAT;
 		bool useShaderDepal = gstate_c.useShaderDepal;
 		bool useSmoothedDepal = gstate_c.useSmoothedShaderDepal;
-		bool colorWriteMask = IsColorWriteMaskComplex(gstate_c.allowFramebufferRead);
+		bool colorWriteMask = pipelineState.maskState.applyFramebufferRead;
```

Comment:
Well, this will enable it a bit more often than necessary, but probably better to have less shader variations.  Won't match the assert below (if a shader blend was needed but write mask is standard.)

-[Unknown]

Reply:
Right, the assert is a little off now. Now that these decisions have been moved to a single point we can make them in a cleaner, more readable way, and I'm going to make it use standard masking when possible (or required due to no shader bitops). Though when anything else reads the framebuffer anyway, we might as well do it in the shader to avoid pipeline variants indeed.